client/control/components/RecipeList.js

```diff
@@ -64,14 +64,61 @@ SwitchFilter.propTypes = {
 };
 
 class DisconnectedRecipeList extends React.Component {
-  propTypes = {
-    dispatch: pt.func.isRequired,
-    isFetching: pt.bool.isRequired,
-    recipeListNeedsFetch: pt.bool.isRequired,
-    recipe: pt.object.isRequired,
-    recipes: pt.array.isRequired,
+  /**
+   * Recipe metadata properties and associated labels to display
+   * @type {Object}
+   */
+  static ActionMetaData = {
+    'show-heartbeat': {
+      // prop name : displayed label
+      surveyId: 'Survey ID',
+    },
+  };
+
+  /**
+   * Given a recipe object, determines what type of recipe it is (based on its `action`),
+   * and then compiles a hash of 'displayed metadata props' and their values. This hash
+   * is saved on the recipe as `metaData`, and displayed in the 'Metadata'
+   *
+   * Beyond that, a string of metaData values is created, and attached to the
+   * recipe as the `searchData` property. This is used by the `Table` component
+   * to search/filter/sort the metaData.
+   *
+   * @param  {Object} Original recipe object
+   * @return {Object} Original recipe but with `metaData` and `searchData` properties added
+   */
+  static applyRecipeMetadata(recipe) {
+    const { action: recipeAction } = recipe;
+    const newRecipe = { ...recipe };
+
+    // recipes should have empty metaData/searchData props,
+    // regardless if we set the values or not
+    newRecipe.metaData = {};
+    newRecipe.searchData = '';
+
+    // check if there are specific properties/labels we want to grub
+    const requestedMetaProps = DisconnectedRecipeList.ActionMetaData[recipeAction];
+
+    // if there are metadata props we want to pull out,
+    // loop through them and extract the data from the recipe
+    for (const propName in requestedMetaProps) {
+      if (requestedMetaProps.hasOwnProperty(propName)) {
+        // get the displayed label value
+        const label = requestedMetaProps[propName];
+        // get the actual property value
+        const value = newRecipe.arguments[propName];
+        // save the label:property into the metadata
+        newRecipe.metaData[label] = value;
+
+        // add the value to the existing search data
+        newRecipe.searchData = `${newRecipe.searchData} ${value}`;
```

Comment:
If we didn't have spaces between values, we could use the simple string concatenation. However, apparently if you attempt to do something like
```
searchData += `${value} `
// or
searchData += value + ' ';
```
it throws a `prefer-template` error while linting (http://eslint.org/docs/rules/prefer-template).

We need the values separated to prevent filters from returning the wrong items. e.g. if the values for an item are `123` and `345`, searching for `33` would still display the item (since the computed value is `123345`)

_SO_ basically I'm going to just revert to what the code originally was, unless you prefer another method